popup/index.tsx

```diff
@@ -1,26 +1,301 @@
-import { useState } from "react"
+import * as ethers from "ethers"
+import { useCallback, useEffect, useState, type MouseEvent } from "react"
+
+import { BackgroundDirectMessenger } from "~packages/messenger/background/direct"
+import { WaalletContentProvider } from "~packages/provider/waallet/content/provider"
 
 import "~style.css"
 
+const provider = new ethers.BrowserProvider(
+  new WaalletContentProvider(new BackgroundDirectMessenger())
+)
+
 function IndexPopup() {
-  const [data, setData] = useState("")
+  const [accountIndex, setAccountIndex] = useState<number>(0)
+  const [accounts, setAccounts] = useState<string[]>([""])
+  const [balances, setBalances] = useState<bigint[]>([0n])
+  const [transactionHashes, setTransactionHashes] = useState<string[]>([""])
+  const [internalTransactionHashes, setInternalTransactionHashes] = useState<
+    string[]
+  >([""])
+  const [explorerUrl, setExplorerUrl] = useState<string>("")
+
+  useEffect(() => {
+    const _accountIndex = 0
```

Comment:
nit: It's not common in TypeScript to prefix naming with underscore. Maybe we could think a proper way to avoid this practice.